src/trace_queue.js

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable prefer-destructuring */
+/* eslint-disable no-console */
+/* eslint-disable valid-jsdoc */
+/* eslint-disable require-jsdoc */
+const EventEmitter = require('events').EventEmitter;
+const axios = require('axios');
+const https = require('https');
+const http = require('http');
+const utils = require('../src/utils.js');
+const config = require('./config.js');
+const consts = require('./consts.js');
+
+
+class TraceQueue extends EventEmitter {
+    constructor(batchSender) {
+        super();
+        this.batchSize = config.getConfig().batchSize;
+        this.batchBytesLimit = consts.batchBytesLimit;
+        this.traces = [];
+        this.batchQueueTime = config.getConfig().batchQueueTime;
+        this.batchSender = batchSender;
+        this.currentByteSize = 0;
+        this.initQueue();
+    }
+
+    /**
+   * Init queue event listners
+   */
+    initQueue() {
+        this.removeAllListeners();
+        this.currentByteSize = 0;
+        this.flush();
+        //  check if queue reached batch size
+        this.on('traceQueued', () => {
+            if (this.batchSizeReached()) {
+                utils.debugLog(`Queue size reached ${this.currentSize}, batch ready! Releasing... `);
+                this.releaseBatch();
+            }
+        });
+
+        // update queue size in bytes + check if limit reached
+        this.on('traceQueued', (trace) => {
+            this.addToCurrentByteSize(trace);
```

Comment:
also better to check here curr size + next instead of adding before the check, to prevent having releaseBatch get a parameter which increases complaxity